src/Button.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef BUTTON_H
+#define BUTTON_H
+#include <Arduino.h>
+#include "original/iot_button.h"
+#include "original/arduino_config.h"
+
+    // Define a function pointer type for callbacks
+    typedef void (*callbackFunction)(void *button_handle, void *usr_data);
+
+    // Define the Button class
+class Button {
+
+public:
+    // Constructors for gpio button
+    Button(int pin, const bool pullup);
+
+    // Constructors for gpio button
+    Button(int pin, const bool pullup, uint8_t adc_channel, uint8_t button_index, uint16_t min, uint16_t max);
```

Comment:
```suggestion
    Button(int pin, bool pullup);

    // Constructors for gpio button
    Button(int pin, bool pullup, uint8_t adc_channel, uint8_t button_index, uint16_t min, uint16_t max);
```
`const` should be used with the pointer.